crates/sui-json-rpc/src/read_api.rs

```diff
@@ -433,3 +526,107 @@ pub fn get_transaction_data_and_digest(
     let txn_digest = TransactionDigest::new(sha3_hash(&intent_msg.value));
     Ok((intent_msg.value, txn_digest))
 }
+
+pub fn get_rendered_fields(
+    fields: VecMap<String, String>,
+    move_struct: &MoveStruct,
+) -> RpcResult<BTreeMap<String, String>> {
+    let sui_move_value: SuiMoveValue = MoveValue::Struct(move_struct.clone()).into();
+    if let SuiMoveValue::Struct(move_struct) = sui_move_value {
+        return fields
+            .contents
+            .iter()
+            .map(|entry| match parse_template(&entry.value, &move_struct) {
+                Ok(value) => Ok((entry.key.clone(), value)),
+                Err(e) => Err(e),
+            })
+            .collect::<RpcResult<BTreeMap<_, _>>>();
+    }
+    Err(anyhow!("Failed to parse move struct"))?
+}
+
+fn parse_template(template: &str, move_struct: &SuiMoveStruct) -> RpcResult<String> {
+    let mut output = template.to_string();
+    let mut var_name = String::new();
+    let mut in_braces = false;
+    let mut escaped = false;
+
+    for ch in template.chars() {
+        match ch {
+            '\\' => {
+                escaped = true;
+                continue;
+            }
+            '{' if !escaped => {
+                in_braces = true;
+                var_name.clear();
+            }
+            '}' if !escaped => {
+                in_braces = false;
+                let value = get_value_from_move_struct(move_struct, &var_name)?;
```

Comment:
Can we do nested paths or is it first level access for now?

Reply:
Nested paths(up to 10) is supported! you can do {{ meta.contact.email }}